src/sentry/static/sentry/app/components/truncate.tsx

```diff
@@ -1,34 +1,42 @@
-import PropTypes from 'prop-types';
 import React from 'react';
+import PropTypes from 'prop-types';
+
+type Props = {
+  value: string;
+  maxLength: number;
+  leftTrim: boolean;
```

Comment:
The `leftTrim` and `maxLength` props are optional in the proptypes, should they be optional here too?

Reply:
If I mark them as optional here, typescript will complain everywhere I use them, that they might be undefined (even though they should not be, as we are setting them in defaultProps). If I leave them required and don't specify them when using <Truncate> somewhere else, it doesn't complain, because it probably detects, that they are being set in defaultProps. What is our general policy, how do we handle this?

Reply:
We have been using the `Partial` utility type to handle default props in a few other components:

* https://github.com/getsentry/sentry/blob/6b75af04c003970a41907c574fbef13281445d3e/src/sentry/static/sentry/app/components/placeholder.tsx#L7-L19
* https://github.com/getsentry/sentry/blob/6b75af04c003970a41907c574fbef13281445d3e/src/sentry/static/sentry/app/components/acl/access.tsx#L53-L75

Reply:
I am afraid that will not change the situation:
![image](https://user-images.githubusercontent.com/9060071/72979126-ed7ee200-3dd7-11ea-9a23-ea4cd2aef2da.png)
Do you think we can keep it like this? https://stackoverflow.com/a/37282264

Reply:
I think I would like to talk about this with team on our next TSC, let's just leave this open for now 👍

Reply:
We have a later version of both the react types and typescript than mentioned in that answer. I wonder if we can remove the Partial defaultProps pattern now that we have upgraded react.

As for this you can ignore my earlier question as it seems default props typing has changed.